### DISCUSSION TITLE
ICommand attribute in 8.0.0-preview4

### DISCUSSION THREAD

**Discussion by imsam67** [CommunityToolkit/dotnet#315] [Q&A] [answered]
The ICommand attribute seems to have disappeared in 8.0.0-preview4. Has it been dropped or renamed or are we to use something different?

**Comment by Sergio0694** [comment, marked as answer]
It's still there, but it's been renamed to `[RelayCommand]` in #288 🙂

**Reply by Ahmadmansoor** [reply]
Ok, that's funny, now all new tuts by the pro guys have been through in, as any the new ppl for Maui will follow the tut and find it will not work.
by the way what happen to AlsoNotifyChangeFor  too?

**Reply by Ahmadmansoor** [reply]
Ok I found it , I write it here just in case someone ask about it too
AlsoNotifyChangeFor >> become >>NotifyPropertyChangedFor

**Comment by imsam67** [comment]
I thought that might be the case but I couldn't get it to work. I'm using this in a .NET MAUI app view model. I use it like this but when I click the button, nothing happens. I put break points to test it and I never hit them.

```
using CommunityToolkit.Mvvm.Input;

public partial class MyViewModel : BaseViewModel
{
     [RelayCommand]
     public async Task DoSomething()
     {
         // Do something here...
     }
}
```

My `BaseViewModel` looks like this:

```
public partial class BaseViewModel : ObservableObject
{
     [ObservableProperty]
     bool isBusy = false;

     [ObservableProperty]
     string title = string.Empty;
 }
```
And in my `ContentPage`, I simply reference the `DoSomething()` method like this:

```
<Button
   Text="Submit"
   Command={Binding DoSomething} />
```

Am I doing somethng wrong here?

**Reply by Sergio0694** [reply]
The generated command will be called "DoSomething**Command**". You can actually see the generated code yourself if you expand the analyzer node in solution explorer.

**Comment by imsam67** [comment]
Great! Thank you!
